### PR TITLE
test(pf5): fix broken tests after PF5 upgrades 

### PR DESCRIPTION
**Pull request by tthvo**
# Welcome to Cryostat! 👋
## Before contributing, make sure you have:
* [x] Read the [contributing guidelines](https://github.com/cryostatio/cryostat/blob/main/CONTRIBUTING.md)
* [x] Linked a relevant issue which this PR resolves
* [x] Linked any other relevant issues, PR's, or documentation, if any
* [x] Resolved all conflicts, if any
* [x] Rebased your branch PR on top of the latest upstream `main` branch
* [x] Attached at least one of the following labels to the PR: `[chore, ci, docs, feat, fix, test]`
* [x] [Signed all commits using a GPG signature](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification#gpg-commit-signature-verification)

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Related to #1303 

### Unit/Integration tests

Fixed broken tests after upgrading to Patternfly 5. This PR involves the following changes to the tests:

- Updated snapshots. Some requires mocking `ref` (See below notes).
- Corrected the display texts or aria-label texts that the tests use for locating the elements.
- Wrapped codes that cause React state changes in `act` where applicable, for example, actions that opens modal.
- Added/Removed tests to accommodate new designs, for example, label editor or range filters.

I also added some tips/notes (i.e. `TESTING.md`) that I figured out when fixing the tests.

### Other bug fixes

Thanks to the unit tests, the following bugs are fixed:

- Fixed bug where in bulk label editor, if a label input is invalid, all labels after that is removed. Only the invalid one should be removed.
- Fixed agent template row action. This was missed in previous PRs.
- Use `SearchInput` for agent template search bar (i.e. has Search icon for consistency).

## Notes

There are some odd instances where snapshot test environment messes up DOM testing with `react-testing-library`. For example, `createNodeMock` (only for snapshot rendering) is leaked into other DOM tests. 

Reordering the snapshot tests to last solved the problem. We might need some restructuring in #661 to split the 2 types of unit tests.

### CODE REVIEW

**Comment by tthvo** [comment]
Pheww! I think the tests are all updated to accommodate the PF5 upgrades now :D

Not satisfied with it though, as many test logics are not yet standardized (i.e. repeated or done differently). I hope there will be some time to get to #848 to clean things up :D

And many thanks again to @maxcao13 for helping out with this work!!!

**Comment by andrewazores** [comment]
I'll start reviewing this now, but it might take a bit of time :-)

File: src/app/TargetView/TargetSelect.tsx [hunk 1]
```diff
@@ -217,7 +216,9 @@ export const TargetSelect: React.FC<TargetSelectProps> = ({ onSelect, simple, ..
             </Dropdown>
           </CardBody>
           <CardExpandableContent>
-            <CardBody>{!selected ? <NoTargetSelected /> : <SerializedTarget target={selected} />}</CardBody>
+            <CardBody id="serialized-target-details">
+              <SerializedTarget target={selected} />
+            </CardBody>
```
Comment (andrewazores) [inline src/app/TargetView/TargetSelect.tsx:221]: I don't think this `<CardExpandableContent>` ever actually gets rendered:

https://www.patternfly.org/components/card#expandable-cards

so maybe it's worth just removing it from here. This looks like the only usage of `<SerializedTarget>`, too.

File: src/test/RecordingMetadata/ClickableLabel.test.tsx [hunk 2]
```diff
@@ -80,63 +77,15 @@ describe('<ClickableLabel />', () => {
...
-    expect(label.classList.contains('pf-m-blue')).toBe(false);
-
-    await user.hover(label);
-
-    const labelStyle = label.style;
-    expect(labelStyle.cursor).toBe('pointer');
-    expect(labelStyle.getPropertyValue('--pf-c-label__content--before--BorderWidth')).toBe('2.5px');
-    expect(labelStyle.getPropertyValue('--pf-c-label__content--before--BorderColor')).toBe(greyLabelBorderColor);
+  it.skip('should display hover effect when hovered', async () => {
+    // styles are now applied from css stylesheets, thus assertions cannot be done
```
Comment (andrewazores) [inline src/test/RecordingMetadata/ClickableLabel.test.tsx:88]: This can probably just be removed.

File: src/app/TargetView/TargetSelect.tsx (hunk 1 above)
Comment (tthvo) [reply src/app/TargetView/TargetSelect.tsx]: Okayy removed :D

**Pull request merged**